### PR TITLE
Update type annotations for BackgroundTask and utils

### PR DESCRIPTION
**Pull request by uriyyo**

### CODE REVIEW

File: starlette/endpoints.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ async def dispatch(self) -> None:
         if is_async:
             response = await handler(request)
         else:
-            response = await run_in_threadpool(handler, request)
+            response = await run_in_threadpool(handler, request)  # type: ignore
```
Comment (Kludex) [inline starlette/endpoints.py]: Maybe this above:
```python
        handler :typing.Callable[[Request], typing.Any] = getattr(self, handler_name, self.method_not_allowed)
```
or just:
```suggestion
            response = await run_in_threadpool(handler, request)  # type: ignore[arg-type]
```

Comment (uriyyo) [reply starlette/endpoints.py]: Thanks, fixed

File: starlette/background.py [hunk 2]
```diff
@@ -1,12 +1,16 @@
 import asyncio
 import typing
 
+import typing_extensions
```
Comment (Kludex) [inline starlette/background.py]: `typing_extensions` is only available for versions below Python 3.8 right now. I guess we'd want to change the setup.py?

Comment (uriyyo) [reply starlette/background.py]: Agree, fixed

Comment (Kludex) [reply starlette/background.py]: But now it needs to verify the version, otherwise it will raise `ModuleNotFoundError` on 3.10.

Comment (uriyyo) [reply starlette/background.py]: Agree, fixed again)

**Review by Kludex — APPROVED** [review]
Thanks @uriyyo ! :tada:

**Pull request merged**